pyproject.toml

```diff
@@ -0,0 +1,151 @@
+    "crontab>=0.21",  # for services.schedule to parse crontab entries
+    "diskcache>=2.8.3",  # for urlcache.DiskCache
+    "h5py",  # for MLHandler saves .h5 files. TODO: Migrate away from .h5
+    "joblib",  # for gramex.ml
+    "lxml",  # for gramex.pptgen2
+    "markdown",  # OPT: for transforms, gramex.services.create_alert()
+    "matplotlib",  # for gramex.data.download() charts, scale.colors, pptgen2
+    "numpy",  # for ml, topcause, pptgen
+    "oauthlib>=1.1.2",  # for socialhandler, twitterstream
+    "openpyxl",  # for gramex.cache.open .XLSX reading
+    "orderedattrdict>=1.6.0",  # for OrderedDict with attr access for configs
+    "packaging",  # for gramex.init() to parse gramex versions
+    "pandas",  # for all data processing
+    "pillow",  # for pptgen2
+    "psutil",  # for gramexadmin to monitor process
+    "python-dateutil",  # for gramex.config.CustomJSONEncoder, gramex.data._convertor
+    # See https://github.com/scanny/python-pptx/issues/754
+    "python-pptx>=0.6.6,<=0.6.19",  # for pptgen2
+    "python-slugify",  # for pre-defined slugs at gramex.config.slug
+    "pyyaml>=5.1",  # for parsing YAML files for config
+    "redis>=2.10.0",  # for RedisStore
+    "requests",  # for all HTTP requests
+    "scikit-learn",  # for MLHandler, gramex.ml. TODO: version >=0.23.2,<1.0
+    "seaborn",  # OPT: gramex.data.download()
+    "six",  # for gramex.yaml backward compatibility
+    "sqlalchemy",  # for gramex.data.filter()
+    "sqlitedict>=1.5.0",  # for SQLiteStore
+    "tables",  # for HDF5 reading / writing. TODO: Where do we need this?
+    "tornado>=5.1.1",  # for Web server
+    "typing_extensions",  # for future-proof typing
+    "watchdog",  # for Monitor file changes
+]
+
+[project.urls]
+Homepage = "https://gramener.com/gramex/"
+Documentation = "https://gramener.com/gramex/guide/"
+Repository = "https://github.com/gramener/gramex"
+Changelog = "https://gramener.com/gramex/guide/release/"
+Issues = "https://github.com/gramener/gramex/issues"
+Stackoverflow = "https://stackoverflow.com/questions/tagged/gramex"
+
+[project.scripts]
+gramex = "gramex:commandline"
+secrets = "gramex.secrets:commandline"
+slidesense = "gramex.pptgen2:commandline"
+
+[tool.setuptools.packages.find]
+include = ["gramex*"]
+
+[project.optional-dependencies]
+# pip install "gramex[full]" for better debugging and ML support
+full = [
+    "boto3",  # for gramex.services.sns.AmazonSNS
+    "datasets",  # for gramex.transformers
+    "ipdb",  # for better debugging
+    "line_profiler",  # for gramex.debug.lineprofile
+    "pymysql",  # for MySQL connections
+    "scipy",  # for gramex.topcause
+    "spacy",  # for gramex.transformers
+    "statsmodels",  # for gramex.ml_api
+    "transformers",  # for gramex.transformers
+    "xlrd",  # for gramex.cache.open .XLS support (not .xlsx)
+    # "conda",  # not required except for rpy2 or reporting conda version
+    # "rpy2",  # deprecated
+]
+influxdb = ["influxdb_client"]
```

Comment:
This is supposed to be `influxdb_client[ciso]`, i.e. we'd install it as `pip install influxdb_client[ciso]` - best reflect that here.

Reply:
Thanks. Added that.